src/org/pentaho/reporting/platform/plugin/cache/DeleteOldOnAccessCache.java

```diff
@@ -120,6 +120,10 @@ protected void setMillisToLive( final long millisToLive ) {
     cleanUp();
   }
 
+  @Override public void cleanupCurrentSession() {
+    cleanup();
```

Comment:
This performs a global cleanup. It does not just limit itself to the current session.
If you want to limit to the current session the list in the "purgeSegment" call below needs to have two entries: SEGMENT and session.getName().